Could not find a declaration file for module 'qs'

### Ziggy version

2.4.1

### Laravel version

10.48.24

### Description

In a Laravel/React/Inertia.js project, we go with the recommendation to get TypeScript types directly from the composer packages, see:

- https://github.com/laravel/breeze/pull/340
- https://github.com/tighten/ziggy?tab=readme-ov-file#typescript

Today, after upgrading to latest and doing `yarn install; yarn build`, we get this in CI/CD

```
vendor/tightenco/ziggy/src/js/index.d.ts:1:26 - error TS7016: Could not find a declaration file for module 'qs'. '/var/www/hr/node_modules/qs/lib/index.js' implicitly has an 'any' type. Try `npm i --save-dev @types/qs` if it exists or add a new declaration (.d.ts) file containing `declare module 'qs';`
```

Or this in local:

```
vendor/tightenco/ziggy/src/js/index.d.ts:1:26 - error TS7016: Could not find a declaration file for module 'qs'. '/Users/dac.chartrand/code/github/Humi-HR/humility/applications/hr/node_modules/qs/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/qs` if it exists or add a new declaration (.d.ts) file containing `declare module 'qs';`

1 import { ParsedQs } from 'qs';
                           ~~~~


Found 1 error in vendor/tightenco/ziggy/src/js/index.d.ts:1
```

### Ziggy call and context

```js
yarn build
```


### Ziggy configuration

```js
// tsconfig.json
{
    "compilerOptions": {
        "allowJs": true,
        "module": "ESNext",
        "moduleResolution": "bundler",
        "jsx": "react-jsx",
        "strict": true,
        "isolatedModules": true,
        "target": "ESNext",
        "esModuleInterop": true,
        "forceConsistentCasingInFileNames": true,
        "noEmit": true,
        "paths": {
            "@/*": ["./resources/js/*"],
            "@images/*": ["./resources/images/*"],
            "ziggy-js": ["./vendor/tightenco/ziggy"]
        }
    },
    "include": ["resources/js/**/*.ts", "resources/js/**/*.tsx", "resources/js/**/*.d.ts"]
}
```


### Route definition

```php
n/a
```

--- a/src/js/index.d.ts
+++ b/src/js/index.d.ts
@@ -1,5 +1,3 @@
-import { ParsedQs } from 'qs';
-
 /**
  * A list of routes and their parameters and bindings.
  *
@@ -158,6 +156,11 @@ interface Config {
         pathname?: string;
         search?: string;
     };
+}
+
+// qs's parsed query params type, so we don't have to have qs as a dependency
+interface ParsedQs {
+    [key: string]: undefined | string | string[] | ParsedQs | ParsedQs[];
 }
 
 /**

